### PR TITLE
16 fix: DHIS2-9198 builds optionSet for in SearchGroupFactory

### PR DESCRIPTION
**Pull request by paschalidi**
Hey 👋  This is the ticket: https://jira.dhis2.org/browse/DHIS2-9198

### CODE REVIEW

File: src/core_modules/capture-core/metaDataMemoryStoreBuilders/common/factory/searchGroup/SearchGroupFactory.js [hunk 1]
```diff
@@ -81,35 +90,52 @@ class SearchGroupFactory {
...
+                translations,
+                displayName,
+                displayShortName,
+                description,
+                unique,
+                valueType,
+            } = searchAttribute.trackedEntityAttribute;
+
+            o.id = id;
+            o.name =
```
Comment (paschalidi) [inline src/core_modules/capture-core/metaDataMemoryStoreBuilders/common/factory/searchGroup/SearchGroupFactory.js:105]: thats of course a personal preference. If you feel that the previous version has some advantages (readability, code quality etc) please let me know and I am happy to discuss/revert the changes. :)

Comment (JoakimSM) [reply src/core_modules/capture-core/metaDataMemoryStoreBuilders/common/factory/searchGroup/SearchGroupFactory.js:105]: I think both are fine. Pretty sure it's more efficient to not do the desctructering in most engines.

File: src/core_modules/capture-core/metaDataMemoryStoreBuilders/common/factory/searchGroup/SearchGroupFactory.js [hunk 2]
```diff
@@ -81,35 +90,52 @@ class SearchGroupFactory {
...
-                programAttribute.renderType && programAttribute.renderType.DESKTOP && programAttribute.renderType.DESKTOP.type);
-        } */
+                optionSet.id,
+                searchAttribute.renderOptionsAsRadio,
+                null,
+                value => value,
+            );
+        }
 
         await Promise.resolve();
```
Comment (JoakimSM) [inline src/core_modules/capture-core/metaDataMemoryStoreBuilders/common/factory/searchGroup/SearchGroupFactory.js:140]: Let's remove this old unnecessary code while we're at it.
```suggestion
```

**Review by JoakimSM — COMMENTED** [review]
Good work 👍 

One bug I found I hope you can have a look at: In an optionSet field, when you deselect the value using the 'x' the app crashes. (For example "Gender" in the "Child Programme")

File: src/core_modules/capture-core/metaDataMemoryStoreBuilders/common/factory/searchGroup/SearchGroupFactory.js (hunk 2 above)
Comment (paschalidi) [reply src/core_modules/capture-core/metaDataMemoryStoreBuilders/common/factory/searchGroup/SearchGroupFactory.js:140]: Good point

**Comment by paschalidi** [comment]
> One bug I found I hope you can have a look at: In an optionSet field, when you deselect the value using the 'x' the app crashes. (For example "Gender" in the "Child Programme")

Tryna reproduce this one and cant see it on my machine. Can you retry see if it is still there?

**Comment by JoakimSM** [comment]
> Tryna reproduce this one and cant see it on my machine. Can you retry see if it is still there?

Seems ok now 👍

**Review by JoakimSM — APPROVED** [review]
👍
